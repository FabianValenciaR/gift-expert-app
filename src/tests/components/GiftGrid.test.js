import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import { GiftGrid } from "../../components/GiftGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componenete <GiftGrid />', () => {
    const category = 'Dragon Ball';

    test('Debe de crearse el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GiftGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/img.jpg',
            title: 'Title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GiftGrid category={category} />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

})
