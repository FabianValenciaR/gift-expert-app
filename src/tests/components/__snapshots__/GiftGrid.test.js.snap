// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componenete <GiftGrid /> Debe de crearse el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
